src/main/java/io/vertx/core/http/WebSocketFrame.java

```diff
@@ -63,6 +63,26 @@ static WebSocketFrame textFrame(String str, boolean isFinal) {
   }
 
   /**
+   * Create a ping WebSocket frame.
+   *
+   * @param data the bytes for the frame
+   * @return the frame
+   */
+  static WebSocketFrame pingFrame(Buffer data) {
+    return factory.pingFrame(data);
+  }
+
+  /**
+   * Create a pong WebSocket frame.
+   *
+   * @param data the bytes for the frame
+   * @return the frame
+   */
+  static WebSocketFrame pongFrame(Buffer data) {
```

Comment:
why is it useful to allow the user to create pong frames if we can't send them ? we should remove this

Reply:
sorry, we allow user to write unsolicited pong frames

Reply:
so never mind on this one :-)